packages/netlify-cms-core/src/actions/entries.ts

```diff
@@ -193,42 +207,28 @@ export function emptyDraftCreated(entry) {
 /*
  * Exported simple Action Creators
  */
-export function createDraftFromEntry(entry, metadata, mediaFiles) {
+export function createDraftFromEntry(entry: EntryMap, metadata?: Map<string, unknown>) {
   return {
     type: DRAFT_CREATE_FROM_ENTRY,
-    payload: { entry, metadata, mediaFiles },
+    payload: { entry, metadata },
   };
 }
 
 export function discardDraft() {
-  return (dispatch, getState) => {
```

Comment:
`mediaFiles` are now under `entryDraft.entry.mediaFiles` so they are cleared when the entry is cleared